package/network/config/netifd/files/lib/netifd/dhcp.script

```diff
@@ -20,7 +20,13 @@ setup_interface () {
 
 	local i
 	for i in $router; do
-		proto_add_ipv4_route "$i" 32 "" "$ip"
+		local ip_net
+		local gw_net
+		ip_net=$(ipcalc.sh $ip/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
+		gw_net=$(ipcalc.sh $i/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
```

Comment:
Usage of grep and awk is awkward; use eval "$(ipcalc.sh $i/$mask)"; gw_net="$NETWORK".